src/snowflake/snowpark/dataframe_writer.py

```diff
@@ -255,6 +256,15 @@ def save_as_table(
                     f"Unsupported table type. Expected table types: {SUPPORTED_TABLE_TYPES}"
                 )
 
+            session = self._dataframe._session
+            if not isinstance(session._conn, MockServerConnection) and save_mode in [
+                SaveMode.APPEND,
+                SaveMode.TRUNCATE,
+            ]:
+                table_exists = session._table_exists(table_name)
```

Comment:
add a comment here about why are we doing the check here

Reply:
added